challenges/frontend/challenge.md

```diff
@@ -0,0 +1,157 @@
+
+Split your code submission using pull requests and give the team an opportunity to review the PRs. A good “rule of thumb” to follow is that the final PR submission is a formality adding a small feature set - it means that the team had an opportunity to contribute the feedback during multiple well defined stages of your work.
+
+Our team will do their best to provide a high quality review of the submitted pull requests in a reasonable time frame. You are spending your time on this, we are going to contribute our time too.
+
+After the final submission, the interview team will assemble and vote using +1, -2 anonymous voting system: +1 is submitted whenever a team member accepts the submission, -2 otherwise.
+In case of a positive result, we will connect you to our HR team who will collect one-two references and will work out other details. You can start the reference collection process in parallel if you would like to speed up the process.
+
+After reference collection, our ops team will send you an offer.
+In case of a negative score result, the hiring manager will contact you and send a list of key observations from the team that affected the result.
+
+## Code and project ownership
+This is a test challenge and we have no intent of using the code you’ve submitted in production. This is your work, and you are free to do whatever you feel is reasonable with it. In the scenario when you don’t pass, you can open source it with any license and use it as a portfolio project.
+
+## Areas of focus
+These are the areas we will be evaluating in the submission:
+
+- Use consistent coding style. We follow [AirBnB Coding Style](https://github.com/airbnb/javascript) for JS and use [prettier](https://prettier.io/) code formatter.
+- Create a few unit-tests for scenarios you think make sense.
+- Make sure builds are reproducible. Pick any vendoring/packaging system that will allow us to get consistent build results.
+- Ensure error handling and error reporting is consistent. The app should report clear errors and not crash under non-critical conditions.
+- Ensure that your app is secure.
+
+## Pitfalls and Gotchas
+To help you out, we’ve composed a list of things that previously resulted in a no-pass from the interview team:
+
+- Scope creep. Candidates have tried to implement too much and ran out of time, energy.
+
+- Avoid using unnecessary 3d party dependencies.
+
+- Error handling. We pay extra attention to error handling. Make sure that they are properly handled and not ignored.
+
+- URL navigation is a big part of great UX, do not ignore it.
+
+- Keep your CSS simple but not simpler. Do not waste your time on animations, instead make sure that an extra character does not destroy your layout.
+
+- Make sure that your code is secured and your application is not vulnerable to CSRF/XSS attacks.
```

Comment:
I'd also add a bit on secure session management and good crypto setup for server side for senior devs.